ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/PartyRestrictionUserManagementAuthIT.scala

```diff
@@ -0,0 +1,114 @@
+          .copy(
+            cacheExpiryAfterWriteInSeconds = UserManagementCacheExpiryInSeconds
+          )
+      )
+    )
+  )
+
+  override def serviceCallName: String = ""
+
+  override protected def serviceCall(context: ServiceCallContext): Future[Any] = ???
+
+  private def createUser(
+      userId: String,
+      serviceCallContext: ServiceCallContext,
+      rights: Vector[uproto.Right] = Vector.empty,
+  ): Future[uproto.CreateUserResponse] = {
+    val user = uproto.User(
+      id = userId,
+      metadata = Some(ObjectMeta()),
+      identityProviderId = serviceCallContext.identityProviderId,
+    )
+    val req = uproto.CreateUserRequest(Some(user), rights)
+    stub(uproto.UserManagementServiceGrpc.stub(channel), serviceCallContext.token)
+      .createUser(req)
+  }
+
+  private def allocateParty(serviceCallContext: ServiceCallContext, party: String) =
+    stub(pproto.PartyManagementServiceGrpc.stub(channel), serviceCallContext.token)
+      .allocateParty(
+        pproto.AllocatePartyRequest(
+          partyIdHint = party,
+          identityProviderId = serviceCallContext.identityProviderId,
+        )
+      )
+
+  it should "allow to grant permissions to parties which are allocated in the IDP" in {
+
+    for {
+      idpConfig <- createConfig(canReadAsAdmin)
+      identityProviderConfig = idpConfig.identityProviderConfig.getOrElse(
+        sys.error("Unable to create IdentityProviderConfig")
+      )
+      (_, idpAdmin) <- createUserByAdmin(
+        userId = UUID.randomUUID().toString + "-alice-1",
+        tokenIssuer = Some(identityProviderConfig.issuer),
+        identityProviderId = identityProviderConfig.identityProviderId,
+        rights = Vector(
+          uproto.Right(
+            uproto.Right.Kind.IdentityProviderAdmin(uproto.Right.IdentityProviderAdmin())
+          ),
+          uproto.Right(uproto.Right.Kind.CanReadAs(uproto.Right.CanReadAs("some-party-2"))),
+        ),
+        primaryParty = "some-party-1",
+      )
+      _ <- createUser(
+        UUID.randomUUID().toString + "-alice-2",
+        idpAdmin,
+      )
+
+      _ <- expectInvalidArgument(
+        createUser(
+          UUID.randomUUID().toString + "-alice-3",
+          idpAdmin,
+          Vector(
+            uproto.Right(uproto.Right.Kind.CanReadAs(uproto.Right.CanReadAs("some-party-1"))),
+            uproto.Right(uproto.Right.Kind.CanActAs(uproto.Right.CanActAs("some-party-2"))),
+          ),
+        )
+      )
+
+      _ <- allocateParty(idpAdmin, "some-party-1")
+      _ <- allocateParty(idpAdmin, "some-party-2")
+
+      _ <- createUser(
+        UUID.randomUUID().toString + "-alice-4",
+        idpAdmin,
+        Vector(
+          uproto.Right(uproto.Right.Kind.CanReadAs(uproto.Right.CanReadAs("some-party-1"))),
+          uproto.Right(uproto.Right.Kind.CanActAs(uproto.Right.CanActAs("some-party-2"))),
+        ),
+      )
+    } yield {
+      assert(true)
+    }
+
+  }
+
+}
```

Comment:
Are you going to add negative cases as well?

Reply:
negative case starts before I allocated parties, so if does not give failed future - the test will fail

Reply:
I see that, actually I meant were tests where admin in IDP1 tries to act on parties in IDP2, is that covered elsewhere?

Reply:
this is covered in authorization tests, the other PR

Reply:
#15994